addons/misra.py

```diff
@@ -1656,6 +1659,148 @@ def get_violations(self):
         return self.violations
 
 
+    def addSuppressedRule(self, ruleNum,
+                          fileName   = None,
+                          lineNumber = None,
+                          symbolName = None):
+        """
+        Add a suppression to the suppressions data structure
+
+        Suppressions are stored in a dictionary of dictionaries that
+        contains a list of tuples.
+
+        The first dictionary is keyed by the MISRA rule in hundreds
+        format. The value of that dictionary is a dictionary of filenames.
+        If the value is None then the rule is assumed to be suppressed for
+        all files.
+        If the filename exists then the value of that dictionary contains the
+        scope of the suppression.  If the value is None then the rule is assumed
+        to be suppresed for the entire file. Otherwise the value of the dictionary
+        is a list of line number, symbol name tuples.
+        For each tuple either line number or symbol name can can be none.
+
+        """
+
+        if lineNumber is not None or symbolName is not None:
+            line_symbol = (lineNumber, symbolName)
+        else:
+            line_symbol = None
+
+        # If the rule is not in the dict already then add it
+        if not ruleNum in self.suppressedRules:
+            ruleItemList = list()
+            ruleItemList.append(line_symbol)
+
+            fileDict = dict()
+            fileDict[fileName] = ruleItemList
+
+            self.suppressedRules[ruleNum] = fileDict
+        else:
```

Comment:
maybe it's not pythonic but I'd like a early return here instead of the `else` to avoid lots of indentation